spec/system/admin/admin_order_deliver_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Order spec', type: :system do
+  before do
+    driven_by(:rack_test)
+    Bullet.raise = false
+  end
+
+  let!(:employee) { create(:employee) }
+  let!(:admin) { create(:admin_user) }
+  let(:company_value) { create(:company_value) }
+  let!(:reward1) { create(:reward, price: 1) }
+  let!(:kudo) { create(:kudo, giver: employee, receiver: employee, company_value: company_value) }
+  let!(:order) { create(:order, employee_id: employee.id, reward: reward1, reward_snapshot: reward1) }
```

Comment:
You can redefine your factories, so they are creating nested objects by themselves - something like: `reward` instead of `reward {}` in Order factory, then you won't have to declare so many objects by yourself - let me know if it is clear, and try refactoring this and other specs